pkg/pullsync/pullsync.go

```diff
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+// Package pullsync provides the pullsync protocol
+// implementation.
```

Comment:
That doesn't say a lot to me. Can you please be more specific on the function of pullsync?

Reply:
see below